src/main/java/br/unb/cic/analysis/df/OverridingAssignmentAnalysis.java

```diff
@@ -31,62 +30,71 @@ public OverridingAssignmentAnalysis(Body methodBody, AbstractMergeConflictDefini
         super(methodBody, definition);
     }
 
+    @Override
+    protected void flowThrough(FlowSet<DataFlowAbstraction> in, Unit u, FlowSet<DataFlowAbstraction> out) {
+        detectConflict(in, u);
+        FlowSet<DataFlowAbstraction> temp = new ArraySparseSet<>();
+
+        FlowSet<DataFlowAbstraction> killSet = new ArraySparseSet<>();
+
+        for (DataFlowAbstraction item : in) {
+            if (statementEquals(item, u)) {
+                killSet.add(item);
+            }
+        }
+        in.difference(killSet, temp);
+        temp.union(gen(u, in), out);
+    }
+
     @Override
     protected FlowSet<DataFlowAbstraction> gen(Unit u, FlowSet<DataFlowAbstraction> in) {
         FlowSet<DataFlowAbstraction> res = new ArraySparseSet<>();
         if (isSourceStatement(u) || isSinkStatement(u)) {
             u.getUseAndDefBoxes().stream().filter(v -> v.getValue() instanceof Local).forEach(v -> {
```

Comment:
I don't understand why this uses getUseAndDefBoxes(), shouldn't this analysis only look for assigned variables? @barbosamaatheus

Reply:
Yes @rafaelmotaalves, you're right.

However, what happens is the following:
![Capture](https://user-images.githubusercontent.com/18462250/86377297-8acf2380-bc5e-11ea-86c2-b16402494d5d.PNG)

I need to get the full array name to add in the abstraction. When I use only **DefBox**, a `JArrayRef `is returned in the value = `arr[3]` In that case I would have to create another auxiliary method to remove the index and still need another specific interaction for arrays.

Using **UseBox** I get a _JimpleLocal_ no value with the name of the array used which simplifies my implementation.

Using only the **UseBox** would be a problem since this same analysis is used for local variables and not just for arrays.

In this sense, using `getUseAndDefBoxes()` solves the problem by attacking both cases with local variables and the case with array.

Can you understand? I am available if you need any more information or have a better idea for that case.

Reply:
Got It, but wouldnt this cause false positives on the case that a variable is used instead of defined, like this?:

```java
int x= 0


x= 1 // source

y = x + 1 // sink

```

Reply:
Hmm, really.
This can be a problem. I will do some tests